control_panel_api/views.py

```diff
@@ -59,13 +83,25 @@ class AppViewSet(viewsets.ModelViewSet):
     filter_fields = ('name', 'repo_url', 'slug')
     permission_classes = (AppPermissions,)
 
+    @transaction.atomic
     def perform_create(self, serializer):
         app = serializer.save(created_by=self.request.user)
-        app.aws_create_role()
 
+        try:
+            app.aws_create_role()
+        except ClientError as e:
```

Comment:
As [said in previous comment](https://github.com/ministryofjustice/analytics-platform-control-panel/pull/67/files#r152300693), we may want to ignore the error raised when "app IAM role already exists".

It's probably less of a problem in the case of app IAM roles as a part from when we deleted the DB I would imagine there shouldn't be apps' IAM roles already there.